cmd/encryption-v1.go

```diff
@@ -438,6 +611,34 @@ func (o *ObjectInfo) IsEncrypted() bool {
 	return false
 }
 
+// IsEncrypted returns true if the object is marked as encrypted.
+func (li *ListPartsInfo) IsEncrypted() bool {
+	if _, ok := li.UserDefined[ServerSideEncryptionIV]; ok {
+		return true
+	}
+	if _, ok := li.UserDefined[ServerSideEncryptionSealAlgorithm]; ok {
+		return true
+	}
+	if _, ok := li.UserDefined[ServerSideEncryptionSealedKey]; ok {
+		return true
+	}
+	return false
+}
```

Comment:
Shouldn't this return true only when all three `ServerSideEncryptionIV`, `ServerSideEncryptionSealAlgorithm` and `ServerSideEncryptionSealedKey` are set? I mean, we're most probably going to fail later in the execution if all these three fields are not set. So why not fail here itself

Reply:
The reason is we are okay to fail later because. Failing later would mean we can reply correct errors to client. Otherwise we will treat it as unencrypted file and return garbage to user.